src/datasets/formatting/np_formatter.py

```diff
@@ -0,0 +1,78 @@
+import sys
+
+import numpy as np
+import pyarrow as pa
+
+from .. import config
+from ..utils.py_utils import map_nested
+from .formatting import Formatter
+
+
+class NumpyFormatter(Formatter[dict, np.ndarray, dict]):
+    def __init__(self, features=None, decoded=True, **np_array_kwargs):
+        super().__init__(features=features, decoded=decoded)
+        self.np_array_kwargs = np_array_kwargs
+
+    def _consolidate(self, column):
```

Comment:
We can use `np.object` for ragged tensors, no?

Reply:
Yes indeed, though I don't know if many users are familiar with np.object numpy arrays, that's shy I switched to using lists.

Not a strong opinion though

Reply:
This is a breaking change, so this is why I think it's better to return `np.object`.

E.g. `Dataset.from_dict({"a": [[1, 2], [3]]}).with_format("np")[:]` returns a numpy array now, but it would return a list with this change.

Reply:
I see, I'll revert this